packages/app/src/app/pages/NewDashboard/Components/Repo/RepoCard.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { Stack, Text, IconButton } from '@codesandbox/components';
+import css from '@styled-system/css';
+import { CardIcon } from './Icons';
+
+export const RepoCard = ({
+  name,
+  path,
+  // interactions
+  selected,
+  onClick,
+  onDoubleClick,
+  onContextMenu,
+  ...props
+}) => (
+  <Stack
+    direction="vertical"
+    gap={2}
+    onClick={onClick}
+    onDoubleClick={onDoubleClick}
+    onContextMenu={onContextMenu}
+    {...props}
+    css={css({
+      width: '100%',
+      height: 240,
+      backgroundColor: 'grays.700',
+      border: '1px solid',
+      borderRadius: 'medium',
+      overflow: 'hidden',
+      borderColor: selected ? 'blues.600' : 'grays.500',
+      ':hover, :focus, :focus-within': {
+        boxShadow: theme => '0 4px 16px 0 ' + theme.colors.grays[900],
+      },
+    })}
+  >
+    <Stack
+      as="div"
+      justify="center"
+      align="center"
+      css={css({
+        height: 160,
+        borderStyle: 'solid',
+        borderWidth: 0,
+        borderBottomWidth: 1,
+        borderColor: 'grays.500',
+        backgroundColor: 'grays.600',
+      })}
+    >
+      <CardIcon />
+    </Stack>
+    <Stack justify="space-between" align="flex-start" marginLeft={4}>
+      <Stack
+        direction="vertical"
+        gap={2}
+        css={css({
+          wordBreak: 'break-all',
+        })}
+      >
+        <Text
+          title={`${props.owner}/${name}/tree/${props.branch}`}
+          size={3}
+          weight="medium"
+          css={css({
+            height: 32,
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            display: '-webkit-box',
+            '-webkit-line-clamp': '2',
+            '-webkit-box-orient': 'vertical',
+          })}
+        >
+          {props.owner}/{name}
+        </Text>
+        <Text size={3} variant="muted" weight="medium">
+          {props.branch}
+        </Text>
+      </Stack>
+      <IconButton
```

Comment:
`IconButton` is really expensive to render for some reason, it takes over 3ms to render at times, which makes the page super slow when scrolling. I have a trick on the `SandboxCard` component where we can prevent the slowdown by just rendering an icon when we're scrolling. It would be good if we could use that here as well!

Reply:
done :)